akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -250,7 +253,10 @@ private[stream] object ConnectionSourceStage {
       .size
 
     private var writeBuffer = ByteString.empty
+
+    // there is data in-flight that we accepted from upstream but haven't successfully written to the connection yet
```

Comment:
"upstream" is a bit confusing for a bidi connection :) Let's maybe call it user and network?

Reply:
Ok, it's upstream and downstream because only half of the stream is implemented using streams... Still confusing but understandable...

Reply:
I agree this was very confusing and it'd be good to change the naming to 'user/stream and network/connection', but this seemed like the current naming and I didn't want to change too much at once ;)

Reply:
Yep, that's fine.